test/e2e/autoscaler_test.go

```diff
@@ -0,0 +1,240 @@
+	})
+})
+
+type ApplyAutoscalerYamlInput struct {
+	E2EConfig    *clusterctl.E2EConfig
+	path         string
+	ClusterProxy framework.ClusterProxy
+}
+
+func ApplyAutoscalerYaml(ctx context.Context, input ApplyAutoscalerYamlInput) {
+	Logf("Applying the cluster autoscaler deployment to the cluster")
+
+	autoscalerYaml, err := os.ReadFile(input.path)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(input.ClusterProxy.Apply(ctx, autoscalerYaml)).To(Succeed())
+
+	Logf("Wait for the autoscaler deployment and collect logs")
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster-autoscaler",        // TODO: get deployment name in input (or infer it from the yaml...)
+			Namespace: "cluster-autoscaler-system", // TODO: get deployment namespace in input (or infer it from the yaml...)
+		},
+	}
+
+	if err := input.ClusterProxy.GetClient().Get(ctx, client.ObjectKeyFromObject(deployment), deployment); apierrors.IsNotFound(err) {
+		framework.WaitForDeploymentsAvailable(ctx, framework.WaitForDeploymentsAvailableInput{
+			Getter:     input.ClusterProxy.GetClient(),
+			Deployment: deployment,
+		}, input.E2EConfig.GetIntervals(bootstrapClusterProxy.GetName(), "wait-controllers")...)
+
+		// Start streaming logs from all controller providers
+		framework.WatchDeploymentLogs(ctx, framework.WatchDeploymentLogsInput{
+			GetLister:  input.ClusterProxy.GetClient(),
+			ClientSet:  input.ClusterProxy.GetClientSet(),
+			Deployment: deployment,
+			LogPath:    filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName(), "logs", deployment.GetNamespace()), // TODO: get log folder in input
+		})
+	}
+}
+
+type AddScaleUpDeploymentAndWaitInput struct {
+	Replicas     int32
+	ClusterProxy framework.ClusterProxy
+}
+
+func AddScaleUpDeploymentAndWait(ctx context.Context, input AddScaleUpDeploymentAndWaitInput, intervals ...interface{}) {
+	Logf("Create a scale up deployment requiring 2G memory for each replica")
+
+	scalelUpDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "scale-up",
+			Namespace: metav1.NamespaceDefault,
+			Labels: map[string]string{
+				"app": "scale-up",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32(input.Replicas),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "scale-up",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "scale-up",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "busybox",
+							Image: "busybox",
+							Resources: corev1.ResourceRequirements{
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceMemory: resource.MustParse("2G"), // TODO: consider if to make this configurable
```

Comment:
this is probably fine for the default tests, we just want to synchronize the requests for the workload with the capacity for the kubemark nodes. by default they are 4GB ram, 2 core cpu. this deployment should be able to fit a single replica per node.